types/js-cookie/index.d.ts

```diff
@@ -86,16 +86,16 @@ declare namespace Cookies {
          * will run the converter first for each cookie. The returned
          * string will be used as the cookie value.
          */
-        withConverter<TConv extends object>(converter: Converter<TConv>): CookiesStatic<TConv>;
+        withConverter<TConv = string>(converter: Converter<TConv>): CookiesStatic<TConv>;
     }
 
-    interface Converter<TConv extends object> {
+    interface Converter<TConv> {
         write?: CookieWriteConverter<TConv> | undefined;
-        read?: CookieReadConverter | undefined;
+        read?: CookieReadConverter<TConv> | undefined;
     }
 
-    type CookieWriteConverter<T extends object> = (value: string | T, name: string) => string;
-    type CookieReadConverter = (value: string, name: string) => string;
+    type CookieWriteConverter<T> = (value: string | T, name: string) => string;
+    type CookieReadConverter<T> = (value: string, name: string) => T | string;
```

Comment:
```suggestion
    type CookieReadConverter<T> = (value: string, name: string) => string | T;
```
just for consistency with 'string | T' used everywhere else.